packages/lib/server/repository/selectedCalendar.ts

```diff
@@ -188,19 +210,43 @@ export class SelectedCalendarRepository {
       // RN we only support google calendar subscriptions for now
       integration: "google_calendar",
       googleChannelExpiration: { not: null },
-      user: {
-        teams: {
-          every: {
-            team: {
-              features: {
-                none: {
-                  featureId: "calendar-cache",
+      AND: [
+        {
+          OR: [
+            // Either is a calendar that has not errored during unwatch
+            { error: null },
+            // Or is a calendar that has errored during unwatch but has not reached max attempts
+            {
+              error: { not: null },
+              unwatchAttempts: {
+                lt: {
+                  // Using ts-ignore instead of ts-expect-error because I am seeing conflicting errors in CI. In one case ts-expect-error fails with `Unused '@ts-expect-error' directive.`
+                  // Removing ts-expect-error fails in another case that _ref isn't defined
+                  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                  // @ts-ignore
+                  _ref: "maxAttempts",
+                  _container: "SelectedCalendar",
+                },
+              },
+            },
+          ],
+        },
+        {
+          user: {
+            teams: {
+              every: {
+                team: {
+                  features: {
+                    none: {
+                      featureId: "calendar-cache",
+                    },
+                  },
```

Comment:
Once we upgrade to prisma v6. This is a good candidate for typed raw sql queries